activejob/test/cases/translation_test.rb

```diff
@@ -0,0 +1,20 @@
+require 'helper'
+require 'jobs/translated_hello_job'
+
+class TranslationTest < ActiveSupport::TestCase
+  setup do
+    JobBuffer.clear
+    I18n.available_locales = [:en, :de]
+    @job = TranslatedHelloJob.new('Johannes')
+  end
+
+  teardown do
+    I18n.available_locales = [:en]
+  end
+
+  test 'it performs the job in the given locale' do
+    @job.locale = :de
+    @job.perform_now
```

Comment:
Yep I agree, the test does a good job, we should keep it. My comment was just about adding a regression test for the case you're describing in your commit message (i.e. with `#perform_later`). Maybe we can add a test (under `test/integration` as pointed by Cristian) along the lines of:

~~~ruby
test 'current I18n.locale is kept running #perform_later' do
  skip if adapter_is?(:inline)

  I18n.locale = :de
  TranslatedHelloJob.perform_later('Johannes')
  wait_for_jobs_to_finish_for(5.seconds)

  assert_equal "Johannes says Guten Tag", JobBuffer.last_value
end
~~~

What do you think ?